docs/Dockerfile

```diff
@@ -10,7 +10,7 @@ COPY Makefile ./
 RUN npm install -g npm@8.1.4 && npm install -g yarn@1.22.17
 RUN make
 
-FROM registry.access.redhat.com/ubi8/nginx-118:1-46
+FROM registry.access.redhat.com/ubi8/nginx-120:latest
```

Comment:
I'm not sure if having latest is the best approach. The result image could be different every time we decide to build (even using same repo version). Maybe better approach is to setup depend bot for this file that will update version when new tag appear?

Reply:
What can go wrong if we depend on `latest`? I can only think of us running into bad image in the repo. Just as well as bad image can hide in `latest` tag, it can be there for more specific one. In either case, we'll need to manually set a previous tag.
Can you think of any scenario that makes non-latest better in any way?

I understand the reasoning behind not deploying the `latest` floating tag but I don't see how it applies to not using `latest` as base image during build.

Reply:
With latest tag as a base image we can end up with different results for same code version. O think it's better to have pinned version as we control when it changes and it make it easier to debug since we will see that change in our commit history. Probably it doesn't apply here as we only copy static files but imagine we have some config, script or lib that fails because some deps has changed.
In other words: I think it's better to pin versions and let dependabot update them than use latest.

Reply:
In general, I agree with the thinking that base image is like a dependency. We pin Go and (JS) dependency versions for a good reason which is reproducible builds.

Just like dependency versions get vulnerabilities assigned over time, container images have that too.
The essential difference is that container consists of many rpm packages, and each one is a dependency that can get security vulnerabilities. The probability of all packages gets summed up, and, therefore, the rate at which **containers** get new vulnerabilities is much higher than the rate at which Go/JS/other language dependencies get vulnerabilities.

In this sense, version update for base containers should be done "constantly" rather than "periodically" like for language dependencies. Realistically, we should always build with effectively-latest releases of base containers.

Frankly, I don't know if dependabot is able to update base container versions and how often it would do that. If you know how to configure it to do that, we can try. Although the downside is that someone would have to approve and merge PRs. This step can be forgotten or not attended for some time which can make us release with outdated base image.

What we do here is actually not much worse than what we do with `main` (see [here](https://github.com/stackrox/stackrox/blob/727774a38c28b1713d710bc491483621b9bd6cb6/image/rhel/Dockerfile#L11)) and other containers where we depend on ubi8-minimal's `8.5` tag. In this case `8.5` is also a **floating** tag and it resolves to different container releases under RHEL/UBI 8.5 family. There too, we can get different results depending on the date and time when we run builds.

Reply:
https://github.com/stackrox/stackrox/pull/1495